File: apps/api/src/main/java/rd/tallerfacil/api/invoice/domain/Invoice.java
package rd.tallerfacil.api.invoice.domain;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import rd.tallerfacil.api.shared.domain.BaseEntity;
import rd.tallerfacil.api.workorder.domain.WorkOrder;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "invoices")
@Getter
@Setter
@NoArgsConstructor
public class Invoice extends BaseEntity {

    @Column(name = "invoice_number", nullable = false, unique = true, length = 20)
    private String invoiceNumber;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "work_order_id", nullable = false)
    private WorkOrder workOrder;

    @Column(name = "issue_date", nullable = false)
    private LocalDate issueDate;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private InvoiceStatus status = InvoiceStatus.PENDIENTE;

    @Column(name = "apply_itbis", nullable = false)
    private boolean applyItbis = false;

    @Column(name = "itbis_rate", nullable = false, precision = 5, scale = 4)
    private BigDecimal itbisRate = new BigDecimal("0.18");

    @Column(nullable = false, precision = 12, scale = 2)
    private BigDecimal subtotal = BigDecimal.ZERO;

    @Column(name = "itbis_amount", nullable = false, precision = 12, scale = 2)
    private BigDecimal itbisAmount = BigDecimal.ZERO;

    @Column(nullable = false, precision = 12, scale = 2)
    private BigDecimal total = BigDecimal.ZERO;

    @Column(columnDefinition = "TEXT")
    private String notes;

    @OneToMany(mappedBy = "invoice", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<InvoiceItem> items = new ArrayList<>();
}

File: apps/web/src/types/invoice.ts
export type InvoiceStatus = "PENDIENTE" | "PAGADA" | "ANULADA";
export type InvoiceItemType = "MANO_OBRA" | "PIEZA";

export interface InvoiceItem {
  id: string;
  item_type: InvoiceItemType;
  description: string;
  quantity: number;
  unit_price: number;
  subtotal: number;
}

export interface Invoice {
  id: string;
  invoice_number: string;
  work_order_id: string;
  vehicle_label: string;
  customer_name: string;
  issue_date: string;
  status: InvoiceStatus;
  apply_itbis: boolean;
  itbis_rate: number;
  subtotal: number;
  itbis_amount: number;
  total: number;
  paid_amount: number;
  remaining_balance: number;
  notes: string | null;
  items: InvoiceItem[];
  created_at: string;
}

export interface InvoicesPage {
  data: Invoice[];
  meta: { total: number; page: number; limit: number };
}

export interface InvoiceItemInput {
  itemType: InvoiceItemType;
  description: string;
  quantity: number;
  unitPrice: number;
}

export interface CreateInvoiceRequest {
  workOrderId: string;
  issueDate: string;
  applyItbis: boolean;
  notes?: string;
  items: InvoiceItemInput[];
}
